test/src/test/java/org/corfudb/runtime/collections/CorfuStoreSecondaryIndexTest.java

```diff
@@ -35,6 +36,46 @@ private CorfuRuntime getTestRuntime() {
         return getDefaultRuntime();
     }
 
+    @Test
+    public void testDisableSecondaryIndexes() throws Exception {
+
+        // Get a Corfu Runtime instance.
+        CorfuRuntime corfuRuntime = getTestRuntime();
+
+        // Creating Corfu Store using a connected corfu client.
+        CorfuStoreShim shimStore = new CorfuStoreShim(corfuRuntime);
+
+        // Define a namespace for the table.
+        final String someNamespace = "some-namespace";
+        // Define table name.
+        final String tableName = "ManagedMetadata";
+
+        { // Positive test.
+            Table<UuidMsg, ExampleValue, ManagedMetadata> table = shimStore.openTable(
+                    someNamespace,
+                    tableName,
+                    UuidMsg.class,
+                    ExampleValue.class,
+                    ManagedMetadata.class,
+                    TableOptions.fromProtoSchema(ExampleValue.class)
+                            .toBuilder().secondaryIndexesDisabled(true).build());
+
+            assertThatExceptionOfType(IllegalArgumentException.class)
+                    .isThrownBy(() -> table.getByIndex("anotherKey", 0L))
+                    .withMessage("Secondary Index anotherKey is not defined.");
+        }
+        { // Negative test.
+            Table<UuidMsg, ExampleValue, ManagedMetadata> table = shimStore.openTable(
+                    someNamespace,
+                    tableName + tableName,
+                    UuidMsg.class,
+                    ExampleValue.class,
+                    ManagedMetadata.class,
+                    TableOptions.fromProtoSchema(ExampleValue.class));
+
+            table.getByIndex("anotherKey", 0L);
```

Comment:
can we warp this in a transaction if possible?

Reply:
Thanks! Fixed.